stl/inc/xutility

```diff
@@ -172,19 +172,28 @@ struct identity {
 template <class _Fx>
 struct _Ref_fn { // pass function object by value as a reference
     template <class... _Args>
-    constexpr decltype(auto) operator()(_Args&&... _Vals) { // forward function call operator
+    constexpr decltype(auto) operator()(_Args&&... _Vals) noexcept(noexcept(_Fn(_STD forward<_Args>(_Vals)...))) {
         return _Fn(_STD forward<_Args>(_Vals)...);
     }
 
     _Fx& _Fn;
 };
 
 template <class _Fn>
-_INLINE_VAR constexpr bool
-    _Pass_functor_by_value_v = sizeof(_Fn) <= sizeof(void*)
-                               && conjunction_v<is_trivially_copy_constructible<_Fn>, is_trivially_destructible<_Fn>>;
+_INLINE_VAR constexpr bool _Pass_functor_by_value_v = conjunction_v<bool_constant<sizeof(_Fn) <= sizeof(void*)>,
+    is_trivially_copy_constructible<_Fn>, is_trivially_destructible<_Fn>>;
 
-template <class _Fn, enable_if_t<_Pass_functor_by_value_v<_Fn>, int> = 0> // TRANSITION, if constexpr
+#if _HAS_IF_CONSTEXPR
+template <class _Fn>
+constexpr auto _Pass_fn(_Fn&& _Val) {
+    if constexpr (_Pass_functor_by_value_v<decay_t<_Fn>>) {
+        return _STD forward<_Fn>(_Val);

```

Comment:
I vaguely recall that `_Pass_fn` matters for debug codegen. If so, should we use `static_cast<_Fn&&>` here (which I am ordinarily not a huge fan of)? Ditto `_Ref_fn` above.

Reply:
Given that `_Pass_fn` is already an annoying "argument-wrapper function", I think it would be a good idea to remove the use of `forward` as another annoying layer of "argument-wrapper function".